File: model/src/main/java/io/sparkled/model/entity/PlaylistSummary.java
package io.sparkled.model.entity;

import java.util.Objects;

public class PlaylistSummary {

    private Integer sequenceCount;
    private Integer durationSeconds;

    public Integer getSequenceCount() {
        return sequenceCount;
    }

    public PlaylistSummary setSequenceCount(Integer sequenceCount) {
        this.sequenceCount = sequenceCount;
        return this;
    }

    public Integer getDurationSeconds() {
        return durationSeconds;
    }

    public PlaylistSummary setDurationSeconds(Integer durationSeconds) {
        this.durationSeconds = durationSeconds;
        return this;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PlaylistSummary that = (PlaylistSummary) o;
        return Objects.equals(sequenceCount, that.sequenceCount) &&
                Objects.equals(durationSeconds, that.durationSeconds);
    }

    @Override
    public int hashCode() {
        return Objects.hash(sequenceCount, durationSeconds);
    }

    @Override
    public String toString() {
        return "PlaylistSummary{" +
                "sequenceCount=" + sequenceCount +
                ", durationSeconds=" + durationSeconds +
                '}';
    }
}

File: model/src/main/kotlin/io/sparkled/model/playlist/PlaylistSummary.kt
package io.sparkled.model.playlist

class PlaylistSummary {

    private var sequenceCount: Int? = null
    private var durationSeconds: Int? = null

    fun getSequenceCount(): Int? {
        return sequenceCount
    }

    fun setSequenceCount(sequenceCount: Int): PlaylistSummary {
        this.sequenceCount = sequenceCount
        return this
    }

    fun getDurationSeconds(): Int? {
        return durationSeconds
    }

    fun setDurationSeconds(durationSeconds: Int): PlaylistSummary {
        this.durationSeconds = durationSeconds
        return this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as PlaylistSummary

        if (sequenceCount != other.sequenceCount) return false
        if (durationSeconds != other.durationSeconds) return false

        return true
    }

    override fun hashCode(): Int {
        var result = sequenceCount?.hashCode() ?: 0
        result = 31 * result + (durationSeconds?.hashCode() ?: 0)
        return result
    }

    override fun toString(): String {
        return "PlaylistSummary(sequenceCount=$sequenceCount, durationSeconds=$durationSeconds)"
    }
}
